microcosm_elasticsearch/searching.py

```diff
@@ -157,14 +145,26 @@ def _order_by(self, query, **kwargs):
         """
         return query.sort("-created_at")
 
-    def _filter(self, query, offset=None, limit=None, **kwargs):
+    def _filter(self, query, offset=None, limit=None, doc_type=None, doc_types=(), **kwargs):
         """
         Filter a query with user-supplied arguments.
 
+        :param doc_types: a list of legal doc types
         :param offset: pagination offset, if any
         :param limit: pagination limit, if any
 
         """
+        if doc_type or doc_types:
+            query = query.filter(
+                "terms",
+                **{
+                    self.doc_type_field: [
+                        doc_type_
```

Comment:
At the moment, this query only works if `doc_type` or `doc_types` contains strings.

I'd like for us to be able to pass our model classes as well. However, we have some inconsistencies in how we define the doc type name so I wasn't able to come up with a conversion I felt good about.